src/main/java/org/micromanager/acqj/internal/Engine.java

```diff
@@ -391,6 +407,13 @@ private void acquireImages(final AcquisitionEvent event,
                core_.setCameraDevice(currentCamera);
             } else {
                core_.snapImage();
+               // note: the definition of SnapImage is that it waits for exposure to
+               // finish.  If it is desired that the AfterCameraHooks can trigger cameras
+               // in Snap mode, those hooks should run in a separate thread, started
+               // after snapImage is started.

```

Comment:
That's a good point, which I'd never considered. I think this probably means after camera hooks should be disallowed when using snap, or a sequence of length 1 should be run if an after camera hook is provided. I think putting it on a separate thread somewhat defeats the purpose. The new camera API should provide a fix to this

Reply:
The other way to do it would be to run SnapImage on a different thread.  Not sure if any changes are needed (other than documenting that you currently can not externally trigger when using Snap).

Reply:
but then you still have a race condition as to whether snap will be called first or an external trigger will be received first, no?

Reply:
Yes, that gets nasty, you would need some mechanism to assure that SnapImage is running before sending the external trigger.  I can only think of ugly ways that will not be completely fool proof.

Reply:
I think its fin to just disallow it for now. The fact that nobody has ever asked about this suggests its not too common of a setup

Reply:
Cool.  I'll leave the comment so that we will not forget.  OK to merge?